prime-router/src/main/kotlin/ReportStreamFilter.kt

```diff
@@ -20,21 +61,183 @@ fun ReportStreamConditionFilter.codes(): List<String> = this.flatMap { it.codes(
     JsonSubTypes.Type(FHIRExpressionConditionFilter::class, name = "fhirExpression"),
 )
 abstract class ConditionFilter(val value: String) {
-    // TODO: change to .filter() or .passes() method and combine with legacy filter
-    abstract fun codes(): List<String>
+    abstract fun evaluate(
+        bundle: Bundle,
+        filter: Boolean = true,
+    ): ConditionFilterResult
+}
+
+abstract class ObservationConditionFilter(value: String) : ConditionFilter(value) {
```

Comment:
The naming here is getting a little confusing to me, what other kind of `*ConditionFilter` are you envisioning?

Reply:
I don't envision any others. As this pattern extends to the rest of the filters I was envisioning `ConditionFilter` becomes `Filter` or `ReportStreamFilter` and `ObservationConditionFilter` becomes `ConditionFilter` or `ObservationFilter`.

The intent was to provide a reusable `evaluate` loop where child classes override `evaluateObservation`.

I think `ObservationFilter` makes the most sense given the above? And then rename the base abstract class after legacy is gone.